spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java

```diff
@@ -0,0 +1,143 @@
+import org.apache.spark.mllib.classification.LogisticRegressionWithLBFGS;
+import org.apache.spark.mllib.feature.HashingTF;
+import org.apache.spark.mllib.regression.LabeledPoint;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Spark Sink plugin that trains a model based upon various labels in the structured record.
+ * Writes this model to a file of a FileSet.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(LogisticRegressionTrainer.PLUGIN_NAME)
+@Description("Uses Logistic Regression to train a model based upon whether messages are spam or not.")
+public class LogisticRegressionTrainer extends SparkSink<StructuredRecord> {
+  public static final String PLUGIN_NAME = "LogisticRegressionTrainer";
+
+  private Config config;
+
+  /**
+   * Configuration for the LogisticRegressionTrainer.
+   */
+  public static class Config extends PluginConfig {
+
+    @Description("The name of the FileSet to save the model to.")
+    private final String fileSetName;
+
+    @Description("Path of the FileSet to save the model to.")
+    private final String path;
+
+    @Description("A comma-separated sequence of fields to use for training.")
+    private final String fieldsToClassify;
+
+    @Description("The field from which to get the prediction. It must be of type double.")
+    private final String predictionField;
+
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
+      " number of features used in LogisticRegressionClassifier. The default value if none is provided will be" +
+      " 100.")
+    private final Integer numFeatures;
+
+    @Nullable
+    @Description("The number of classes to use in training the model. It must be of type integer. " +
+      "The default value if none is provided will be 2.")
+    private final Integer numClasses;
+
+    public Config(String fileSetName, String path, String fieldsToClassify, String predictionField,
+                  Integer numFeatures, Integer numClasses) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.fieldsToClassify = fieldsToClassify;
+      this.predictionField = predictionField;
+      this.numFeatures = numFeatures;
+      this.numClasses = numClasses;
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      Schema.Type predictionFieldType = inputSchema.getField(config.predictionField).getSchema().getType();
+      Preconditions.checkArgument(predictionFieldType == Schema.Type.INT,
+                                  "Prediction field must be of type Int, but was %s.", predictionFieldType);
+    }
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception {
```

Comment:
`SparkSink` does not implement `prepareRun` and it's abstract in `BatchConfigurable` so it has to be implemented here.